% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/default_params_doc.R
\name{default_params_doc}
\alias{default_params_doc}
\title{This function does nothing. It is intended to inherit is parameters'
documentation.}
\usage{
default_params_doc(
  download_url,
  fasta_filename,
  fasta_file_text,
  folder_name,
  protein_sequence,
  protein_sequences,
  pureseqtm_filename,
  pureseqtm_proteome_text,
  pureseqtm_result,
  pureseqtm_url,
  temp_fasta_filename,
  temp_folder_name,
  tmhmm_topology,
  topology,
  topology_filename,
  topology_str,
  topology_strs,
  verbose
)
}
\arguments{
\item{download_url}{the URL to download PureseqTM from}

\item{fasta_filename}{path to a FASTA file}

\item{fasta_file_text}{text of a FASTA file}

\item{folder_name}{superfolder of PureseqTM.
The superfolder's name is \code{/home/[user_name]/.local/share}
by default, as can be obtained by
\link{get_default_pureseqtm_folder}}

\item{protein_sequence}{a protein sequence, with
the amino acids as capitals, for
example \code{MEILCEDNTSLSSIPNSL}.
Use \link{check_protein_sequence} to check if
a protein sequence is valid.}

\item{protein_sequences}{one ore more protein sequence,
each sequence with the amino acids as capitals, for
example \code{MEILCEDNTSLSSIPNSL}}

\item{pureseqtm_filename}{filename to write the PureseqTM results to}

\item{pureseqtm_proteome_text}{the output of a call
to \code{PureseqTM_proteome.sh}}

\item{pureseqtm_result}{the result of a PureseqTM run}

\item{pureseqtm_url}{URL of the PureseqTM git repository}

\item{temp_fasta_filename}{temporary FASTA filename,
which will deleted after usage}

\item{temp_folder_name}{path of a temporary folder.
The folder does not need to exist.
Files that are out in this folder are not automatically
deleted, which is not a problem, as the default
path given by \link{tempdir} is automatically cleaned
by the operating system}

\item{tmhmm_topology}{topology as used by \code{TMHMM}}

\item{topology}{the topology as a \link[tibble]{tibble}
with the columns 'name' and 'topology',
where the 'name' column hold all the proteins' names,
and 'topology' contains the respective topologies as strings.}

\item{topology_filename}{name of the file to save a protein's
topology to}

\item{topology_str}{the topology as a string,
for example \code{000000111100000}}

\item{topology_strs}{the topologies as zero, one oor more strings,
for example \code{c("0", "1")}}

\item{verbose}{set to TRUE for more output}
}
\description{
This function does nothing. It is intended to inherit is parameters'
documentation.
}
\note{
This is an internal function, so it should be marked with
\code{@noRd}. This is not done, as this will disallow all
functions to find the documentation parameters
}
\author{
Rich√®l J.C. Bilderbeek
}
